### PR TITLE
[release] Update changelog for 0.2.23

### PR DESCRIPTION
**Pull request by tpoterba**

### CODE REVIEW

**Comment by danking** [comment]
Would it work if we used "Version 0.2.23", I think that looks visually fine.

**Comment by tpoterba** [comment]
done

File: hail/python/hail/docs/change_log.md [hunk 1]
```diff
@@ -1,15 +1,40 @@
...
+### New features
+- (hail#7009) Introduced analysis pass in Python that mostly obviates the `hl.bind` and `hl.rbind` operators; idiomatic Python that generates Hail expressions will perform much better.
+- (hail#7076) Improved memory management in generated code, add additional log statements about allocated memory to improve debugging.
+- (hail#7085) Warn only once about schema mismatches during JSON import (used in VEP, Nirvana, and sometimes `import_table`.
+- (hail#7106) `hl.agg.call_stats` can now accept a number of alleles for its `alleles` parameter, useful when dealing with biallelic calls without the alleles array at hand.
+ 
+### Performance
+- (hail#7086) Improved performance of JSON import.
+- (hail#6981) Improved performance of Hail min/max/mean operators. Improved performance of `split_multi_hts` by an additional 33%.
+- (hail#7082)(hail#7096)(#7098) Improved performance of large pipelines involving many `annotate` calls.
```
Comment (danking) [inline hail/python/hail/docs/change_log.md]: does that guy need a `hail` before the `#`?

Comment (tpoterba) [reply hail/python/hail/docs/change_log.md]: yep

Comment (danking) [inline hail/python/hail/docs/change_log.md]:
```suggestion
- (hail#7082)(hail#7096)(hail#7098) Improved performance of large pipelines involving many `annotate` calls.
```

Comment (tpoterba) [reply hail/python/hail/docs/change_log.md]: done

Comment (tpoterba) [reply hail/python/hail/docs/change_log.md]: oh, oops, just made the change.

**Comment by danking** [comment]
looks good to me

**Pull request merged**